packages/app/src/sandbox/eval/transpilers/babel/index.ts

```diff
@@ -46,96 +51,96 @@ class BabelTranspiler extends WorkerTranspiler {
-       * in non-strict mode. Transpiling this code would add a "use strict;" piece, which
-       * would then break the code (because it expects `this` to be global). No transpiler
-       * can fix this, and because of this we need to just specifically ignore this file.
-       */
-      if (path === '/node_modules/babel-standalone/babel.js') {
-        resolve({ transpiledCode: code });
-        return;
-      }
+  ): Promise<TranspilationResult> {
+    const { path } = loaderContext;
+    const isNodeModule = path.startsWith('/node_modules') || isUrl(path);
+
+    /**
+     * We should never transpile babel-standalone, because it relies on code that runs
+     * in non-strict mode. Transpiling this code would add a "use strict;" piece, which
+     * would then break the code (because it expects `this` to be global). No transpiler
+     * can fix this, and because of this we need to just specifically ignore this file.
+     */
+    if (path === '/node_modules/babel-standalone/babel.js') {
+      return { transpiledCode: code };
+    }
 
-      // Check if we can take a shortcut, we have a custom pipeline for transforming
-      // node_modules to commonjs and collecting deps
-      if (loaderContext.options.simpleRequire || isNodeModule) {
-        try {
-          const ast: ESTreeAST = parseModule(code);
-          if (isESModule(code)) {
-            measure(`esconvert-${path}`);
-            convertEsModule(ast);
-            endMeasure(`esconvert-${path}`, { silent: true });
-          }
+    // Check if we can take a shortcut, we have a custom pipeline for transforming
+    // node_modules to commonjs and collecting deps
+    if (loaderContext.options.simpleRequire || isNodeModule) {
+      try {
+        const ast: ESTreeAST = parseModule(code);
+        if (isESModule(code)) {
+          measure(`esconvert-${path}`);
+          convertEsModule(ast);
+          endMeasure(`esconvert-${path}`, { silent: true });
+        }
 
-          const syntaxInfo = getSyntaxInfoFromAst(ast);
-          // If the code is commonjs and does not contain any more jsx, we generate and return the code.
-          if (!syntaxInfo.jsx && !syntaxInfo.esm) {
-            measure(`dep-collection-${path}`);
-            collectDependencies(ast).forEach(dependency => {
+        const syntaxInfo = getSyntaxInfoFromAst(ast);
+        // If the code is commonjs and does not contain any more jsx, we generate and return the code.
+        if (!syntaxInfo.jsx && !syntaxInfo.esm) {
+          measure(`dep-collection-${path}`);
+          await Promise.all(
+            collectDependencies(ast).map(async dependency => {
               if (dependency.isGlob) {
                 loaderContext.addDependenciesInDirectory(dependency.path);
               } else {
-                loaderContext.addDependency(dependency.path);
+                await loaderContext.addDependency(dependency.path);
```

Comment:
What's the advantage of awaiting a promise in this case for adding a dependency?

Reply:
I think it was causing race conditions as we utilise the dependencies in `TranspiledModule#doTranspile` which only really became obvious due to the extra delay fetching an ESModule adds

Reply:
Just tested this a little bit and didn't see any noticeable performance impact:

https://codesandbox.io/s/muddy-lake-zpstd
Before ESM: 1.8-2.15secs
After ESM: 1.9-2.1 secs

https://codesandbox.io/s/largish-redux-example-pjc8f
Before ESM: 2.5-2.9secs
After ESM: 2.3-2.6secs

Reply:
Ohhh okay, great! Let's keep this in then :shipit: